app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -401,6 +399,8 @@ protected void showLocaleChangeMenu(final CommCareActivityUIController uiControl
             }
             rebuildOptionsMenu();
             dismissAlertDialog();
+            // recreate activity because context could be changed for locale
+            recreate();
```

Comment:
Don't think it's no longer necessary since we are not changing directions based on App language now.